datasource/virtualmachine/filters.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package virtualmachine
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/packer-plugin-vsphere/datasource/common/driver"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/mo"
+)
+
+// filterByNameRegex filters machines by matching their names against defined regular expression.
+func filterByNameRegex(vmList []*object.VirtualMachine, nameRegex string) []*object.VirtualMachine {
```

Comment:
Those functions all seem to perform very similarly, I wonder if we can factorise them a bit.
A common idiom in Go for a pattern like this is to use unit functions on each item from the list as a filter, I believe you @castorsky did something similar on the Proxmox plugin not too long ago: https://github.com/hashicorp/packer-plugin-proxmox/pull/295/files#diff-c76b98221848287583a48bc198806c52866f92463a335ae3d06959d967e23783R257
Unless it's not possible in this case, I would suggest adopting a similar approach for this PR as well.

Reply:
Done. Covered with [4bc1cfa](https://github.com/hashicorp/packer-plugin-vsphere/pull/498/commits/4bc1cfaa8c45b7a164aea3ee10db4aa17e665b30)